src/drivers/intel/cavs/mn.c

```diff
@@ -52,18 +53,20 @@ struct mn {
 	spinlock_t lock; /**< lock mechanism */
 };
 
-struct mn mn;
+static SHARED_DATA struct mn mn;
 
 void mn_init(struct sof *sof)
 {
 	int i;
 
-	sof->mn = &mn;
+	sof->mn = cache_to_uncache(&mn);
```

Comment:
I assume this platform specific way is fine as it's common to all of cavs? At least it's platform specific code.

Also, is it really needed given that the structure is in shared memory? Is shared memory even cached?

Reply:
Isn't there a platform_shared_get or similar function which you introduced?

Reply:
@paulstelian97 Yes, I guessed it can be used here. I chose not to, because it's cAVS specific driver.